tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/cloudcredentials"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/shepherd/pkg/wait"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	localCluster   = "local"
+	fleetNamespace = "fleet-default"
+)
+
+type ClusterTemplateTest struct {
+	Repo             *v1.ClusterRepo `json:"repo,omitempty" yaml:"repo,omitempty"`
+	TemplateName     string          `json:"templateName,omitempty" yaml:"templateName,omitempty"`
+	TemplateProvider string          `json:"templateProvider,omitempty" yaml:"templateProvider,omitempty"`
+}
+
+type ClusterTemplateTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	templateConfig     *ClusterTemplateTest
+	cloudCredentials   *cloudcredentials.CloudCredential
+}
+
+func (r *ClusterTemplateTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *ClusterTemplateTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+	r.templateConfig = new(ClusterTemplateTest)
+	config.LoadConfig("templateTest", r.templateConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+	r.client = client
+
+	enabled := true
+	var testuser = namegenerator.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	provider := provisioning.CreateProvider(r.templateConfig.TemplateProvider)
+	r.cloudCredentials, err = provider.CloudCredFunc(client)
+	require.NoError(r.T(), err)
+}
+
+func (r *ClusterTemplateTestSuite) TestProvisionRKE2TemplateCluster() {
+	_, err := r.client.Catalog.ClusterRepos().Create(context.TODO(), r.templateConfig.Repo, metav1.CreateOptions{})
```

Comment:
For additional layer of abstraction, I would argue that this should go in a file called `templates.go` in the `rke2` package and you just call the function. If we can help it, we should hide this level of stuff in the test itself.

Reply:
My 2 cents, since this is a simple create call. I like that Sam did it this way, our session cleanup will take care of it anyways. The test also needs to be clear what's going on. I say leave it.

Reply:
As per discussions in this PR and a few through other channels I have created a few generic functions for cluster repos in shepherd